src/index.js

```diff
@@ -6,37 +6,23 @@ const validation = {
 
     init() {
 
-        $('[data-validate] input').on('keyup paste', function() {
+        $('[data-validate] input').on('keyup paste blur', function() {
 
-            const $input = $(this);
-
-            if (validateInput($input).isEmpty()) {
-                clearInputError($input);
-            }
+            updateInputError($(this));
 
         });
 
-        $('[data-validate] input').on('blur', function() {
-
-            const $input = $(this);
-
-            updateInputError($input, validateInput($input));
-
-        });
 
         $('[data-validate]').on('submit', function(e) {
 
             const formErrors = new ErrorBag();
 
             $('input', $(this)).each(function () {
 
-                const $input = $(this);
-                const errors = validateInput($input);
+                const errors = updateInputError($(this));
 
                 formErrors.merge(errors);
 
```

Comment:
Reasoning for removing this? It handles the case of clicking submit on an empty form.